linode/domain/framework_models.go

```diff
@@ -1,42 +1,102 @@
+	RefreshSec  customtypes.LinodeDomainSecondsValue `tfsdk:"refresh_sec"`
+	MasterIPs   types.Set                            `tfsdk:"master_ips"`
+	AXFRIPs     types.Set                            `tfsdk:"axfr_ips"`
+	Tags        types.Set                            `tfsdk:"tags"`
+}
+
+func domainDeepEqual(plan, state DomainModel) bool {
+	return state.Domain.Equal(plan.Domain) &&
+		state.Type.Equal(plan.Type) &&
+		state.Group.Equal(plan.Group) &&
+		state.Status.Equal(plan.Status) &&
+		state.Description.Equal(plan.Description) &&
+		state.SOAEmail.Equal(plan.SOAEmail) &&
+		state.TTLSec.Equal(plan.TTLSec) &&
+		state.RetrySec.Equal(plan.RetrySec) &&
+		state.ExpireSec.Equal(plan.ExpireSec) &&
+		state.RefreshSec.Equal(plan.RefreshSec) &&
+		state.MasterIPs.Equal(plan.MasterIPs) &&
+		state.AXFRIPs.Equal(plan.AXFRIPs) &&
+		state.Tags.Equal(plan.Tags)
 }
 
-func (m *DomainModel) parseDomain(domain *linodego.Domain) {
+func (m *DomainModel) parseComputed(
+	ctx context.Context,
+	domain *linodego.Domain,
+) diag.Diagnostics {
 	m.ID = types.Int64Value(int64(domain.ID))
-	m.Domain = types.StringValue(domain.Domain)
-	m.Type = types.StringValue(string(domain.Type))
-	m.Group = types.StringValue(domain.Group)
-	m.Status = types.StringValue(string(domain.Status))
-	m.Description = types.StringValue(domain.Description)
-	m.MasterIPs = helper.StringSliceToFramework(domain.MasterIPs)
-	m.AXFRIPs = helper.StringSliceToFramework(domain.AXfrIPs)
-	m.TTLSec = types.Int64Value(int64(domain.TTLSec))
-	m.RetrySec = types.Int64Value(int64(domain.RetrySec))
-	m.ExpireSec = types.Int64Value(int64(domain.ExpireSec))
-	m.RefreshSec = types.Int64Value(int64(domain.RefreshSec))
-	m.SOAEmail = types.StringValue(domain.SOAEmail)
-	m.Tags = helper.StringSliceToFramework(domain.Tags)
+	m.TTLSec = customtypes.LinodeDomainSecondsValue{
+		Int64Value: types.Int64Value(int64(domain.TTLSec)),
+	}
+	m.RetrySec = customtypes.LinodeDomainSecondsValue{
+		Int64Value: types.Int64Value(int64(domain.RetrySec)),
+	}
+	m.ExpireSec = customtypes.LinodeDomainSecondsValue{
+		Int64Value: types.Int64Value(int64(domain.ExpireSec)),
+	}
+	m.RefreshSec = customtypes.LinodeDomainSecondsValue{
+		Int64Value: types.Int64Value(int64(domain.RefreshSec)),
+	}
+
+	masterIPs, diags := basetypes.NewSetValueFrom(ctx, types.StringType, domain.MasterIPs)
+	if diags.HasError() {
+		return diags
+	}
+	m.MasterIPs = masterIPs
+
+	axfrIPs, diags := basetypes.NewSetValueFrom(ctx, types.StringType, domain.AXfrIPs)
+	if diags.HasError() {
+		return diags
+	}
+	m.AXFRIPs = axfrIPs
+
+	var tags basetypes.SetValue
+	if len(domain.Tags) == 0 {
+		tags = types.SetNull(types.StringType)
+	} else {
+		tags, diags = basetypes.NewSetValueFrom(ctx, types.StringType, domain.Tags)
+		if diags.HasError() {
+			return diags
+		}
+	}
+	m.Tags = tags
+	return nil
+}
+
+func (m *DomainModel) parseNonComputed(
+	ctx context.Context,
+	domain *linodego.Domain,
+) diag.Diagnostics {
```

Comment:
We don't have to return diagnostics from this function since there isn't any error raised from the parsing.